components/eamxx/src/control/CMakeLists.txt

```diff
@@ -13,7 +35,7 @@ set(SCREAM_CONTROL_HEADERS
 )
 
 add_library(scream_control ${SCREAM_CONTROL_SOURCES})
-target_link_libraries(scream_control PUBLIC scream_share scream_io)
+target_link_libraries(scream_control PUBLIC scream_share scream_io ${dynLibName})
```

Comment:
If I understand correctly, you need the dyn lib for `fv_phys_rrtmgp_active_gases_init` and `fv_phys_rrtmgp_active_gases_set_restart`, right? Those free functions do not seem to have any strict dependency on the dyn lib, so maybe we should move those to something like `share/util/eamxx_fv_phys_rrtmgp_active_gases_workaround.*pp`, along with the type `TraceGasesWorkaround`. Then, we would not have to link dyn to the control lib.

Reply:
We could make TraceGasesWorkaround a singleton, so that it's accessible from both the `fv_phys_rrtmgp_active_gases_xyz` routines, as well as from HommeDynamics.

Reply:
Yeah, this is a good idea. I'll work on it.